song-server/src/main/java/bio/overture/song/server/model/dto/Payload.java

```diff
@@ -21,9 +20,31 @@
 @EqualsAndHashCode(callSuper = true)
 @JsonInclude(value = JsonInclude.Include.NON_NULL)
 public class Payload extends DynamicData {
+  private static final ObjectMapper MAPPER = payloadObjectMapper();
 
   private String studyId;
   private AnalysisTypeId analysisType;
   private List<CompositeEntity> samples;
   private List<FileEntity> files;
+
+  @SneakyThrows
+  public static Payload parse(JsonNode jsonStr) {
+    // convert to hashMap
+    val jsonNode = MAPPER.readValue(jsonStr.toString(), HashMap.class);
+    // writeValueAsString will remove null/empty values from map, including nested
+    val sanitized = MAPPER.writeValueAsString(jsonNode);
```

Comment:
The reason `writeValueAsString` will remove null/empty values is because the `MAPPER`  includes non-null non-empty values as defined in `payloadObjectMapper` method?

Reply:
Yes, that is correct. I guess this comment can be more specific abotu that.